Sources/LeafKit/LeafSerializer.swift

```diff
@@ -100,7 +100,29 @@ struct LeafSerializer {
     }
     
     mutating func serialize(_ loop: Syntax.Loop) throws {
-        guard let array = data[loop.array]?.array else { throw "expected array at key: \(loop.array)" }
+        let finalData: [String: LeafData]
+        let pathComponents = loop.array.split(separator: ".")
```

Comment:
I think there's a bit of code being repeated here for accessing a variable. See the code here: https://github.com/vapor/leaf-kit/blob/master/Sources/LeafKit/LeafSerializer.swift#L91-L97

Maybe we could pull that out into a method for digging into a variable? I'd be surprised if that didn't already exist though.

Reply:
I don't think there's one in there now, if we add more we can break this one out 👍